src/boot/efi/util.c

```diff
@@ -527,3 +527,43 @@ VOID clear_screen(UINTN attr) {
         uefi_call_wrapper(ST->ConOut->SetAttribute, 2, ST->ConOut, attr);
         uefi_call_wrapper(ST->ConOut->ClearScreen, 1, ST->ConOut);
 }
+
+VOID draw_box(UINTN x, UINTN y, UINTN w, UINTN h, UINTN color) {
+        print_at(x, y, color, L"┌");
+        print_at(x + w, y, color, L"┐");
+        print_at(x, y + h, color, L"└");
+        print_at(x + w, y + h, color, L"┘");
+
+        for (UINTN i = 1; i < w; i++) {
+                print_at(x + i, y, color, L"─");
+                print_at(x + i, y + h, color, L"─");
+        }
+
+        for (UINTN i = 1; i < h; i++) {
+                print_at(x, y + i, color, L"│");
+                print_at(x + w, y + i, color, L"│");
+        }
+}
+
+BOOLEAN boxdraw_supported(VOID) {
+        EFI_STATUS err;
+        EFI_SERIAL_IO_PROTOCOL *ser = NULL;
+        static BOOLEAN checked = FALSE, supported = FALSE;
```

Comment:
we usually use "tristate" variables for things like this. i.e. signed integers that carry < 0 for unset, 0 for false, and > 0 for true. I think it would make sense to follow this pattern here too, i.e. do `static INTN supported = -1;` and then check with < 0 if unset, and afterwards pretend it as a regular boolean.